src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnection.cs

```diff
@@ -462,8 +462,28 @@ private void CloseInnerConnection()
 
         override public void Close()
         {
+            ConnectionState previousState = State;
+            Guid operationId;
+            Guid clientConnectionId;
+
+            // during the call to Dispose() there is a redundant call to 
```

Comment:
I am in favor of the diagnostics code checking for the state of connection without really short circuiting the rest of the behavior.
@tstringer as discussed, can you change the condition to check if the ConnectionState.Closed != previousState to check for non-closed connection? In case of non-closed connections, there is a possibility of the connection retaining the Guid.